src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -0,0 +1,156 @@
+  <Target Name="ExtractProjectReferences">
+    <ItemGroup Condition="'$(KeepAllProjectReferences)' != 'true'">
+        <!-- Don't alter the ProjectReference's from the project during the extraction phase -->
+        <_ClonedProjectReference Include="@(ProjectReference)" />
+        <_RemovePackageProjectReference Condition="'%(_ClonedProjectReference.KeepProjectReference)' != 'true' and !$([System.Text.RegularExpressions.Regex]::IsMatch('%(_ClonedProjectReference.FullPath)', '\\tests\\', System.Text.RegularExpressions.RegexOptions.IgnoreCase))"
+                                  Include="%(_ClonedProjectReference.Identity)" />
+        <!-- Populate items first with the project references so that we can trim them but keep the item metadata -->
+        <_PackageProjectReference Include="@(_ClonedProjectReference)" />
+        <_ClonedProjectReference Remove="@(_RemovePackageProjectReference)" />
+        <_PackageProjectReference Remove="@(_ClonedProjectReference)" />                                
+    </ItemGroup>
+  </Target>
+
+  <!-- Given a package drop, or list of package drops, retrieve a list of assembly names and their associated versions -->
+  <Target Name="GatherAssemblyVersionsFromPackagesDrops">
+    <GatherAssemblyVersionsFromPackageDrop PackageDrops="@(_PackagesDrops)">
+       <Output TaskParameter="AssemblyNames" ItemName="_AssemblyVersionList" />                                                  
+    </GatherAssemblyVersionsFromPackageDrop>                                           
+  </Target>
+
+  <Target Name="GatherProjectReferenceForProjectJson"
+          Condition="'$(KeepAllProjectReferences)' != 'true'"
+          Inputs="%(_PackageProjectReference.Identity)"
+          Outputs="fake">
+    <PropertyGroup>
+      <_PackageReferenceName>%(_PackageProjectReference.PackageReferenceName)</_PackageReferenceName>
+      <_InjectProjectReference>%(_PackageProjectReference.FileName)</_InjectProjectReference>
+
+      <!-- determine the version of the assembly in the project reference by looking at the packages in "PackagesDrops".  -->
+      <!-- If an alternate packagereferencename was specified, then look for that version number instead of the version number for the project reference. -->
+      <_AssemblyVersion Condition="'$(_PackageReferenceName)' != '' and '%(_AssemblyVersionList.AssemblyName)' == '$(_PackageReferenceName)'">%(_AssemblyVersionList.AssemblyVersion)</_AssemblyVersion>
+      <_AssemblyVersion Condition="'$(_AssemblyVersion)' == '' and '%(_AssemblyVersionList.AssemblyName)' == '$(_InjectProjectReference)'">%(_AssemblyVersionList.AssemblyVersion)</_AssemblyVersion>
+    </PropertyGroup>
+
+    <Message Condition="'$(_AssemblyVersion)' != ''" Text="Derived version number from packages drop." />   
+
+    <ItemGroup>
+      <_AdditionalProperties Include="%(_PackageProjectReference.Identity)">
+        <AssemblyVersionTransition>$(AssemblyVersionTransition)</AssemblyVersionTransition>
```

Comment:
@ericstj sets this in dir.props as a temporary change (https://github.com/dotnet/corefx/commit/0f268ed1079c182d71cdf49cde3f339a01d67863).  Not providing this parameter caused a break because we needed to ensure we were evaluating the correct AssemblyVersion.  At some point this will just be dead code and need to be cleaned up when Eric remove's that change.